### PR TITLE
4 feature 6 implement death detector and win checker

### PR DESCRIPTION
**Pull request by Kaifeng-Li**
Death Detector and Winloss Checker Finished 
Test passed

### CODE REVIEW

**Comment by ethanhsu8004** [comment]
for ProcessPlayer method, in line 27-32, will it only remove 1 card at most? In other words, can there only exist one creature that has <= 0 healthpoints? If not, there could be some bugs that arise.

File: src/main/java/use_cases/death_detector_use_case/DeathDetectorInteractor.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    }
+
+    private void processPlayer(Player player1, Player player2, List<Integer> ids) {
+        List<Integer> defeatCreatureIds = new ArrayList<>();
+        for (CreatureCard c : player1.getCreatures()) {
+            if (c.getHitPoints() <= 0) {
+                defeatCreatureIds.add(c.getId());
+                player1.removeCreature(c.getId());
+            }
+        }
```
Comment (Kevin22888) [inline src/main/java/use_cases/death_detector_use_case/DeathDetectorInteractor.java:32]: @ethanhsu8004 raised a good point, I also commented on the other pull request with the same problem. His solution is to use an iterator, but this is more readable.

```suggestion
        List<CreatureCard> deadCreatures = new ArrayList<>();
        for (CreatureCard c : player1.getCreatures()) {
            if (c.getHitPoints() <= 0) {
                defeatCreatureIds.add(c.getId());
                deadCreatures.add(c);
            }
        }
        player1.getCreatures().removeAll(deadCreatures);
```

File: src/main/java/use_cases/death_detector_use_case/DeathDetectorResponseModel.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+public class DeathDetectorResponseModel {
+    private List<Integer> defeatCreatureId;
+    private List<Integer> p1CreatureIds;
+    private List<Integer> p2CreatureIds;
+    private List<Integer> p1CreatureHitPoints;
+    private List<Integer> p2CreatureHitPoints;
+    private List<Integer> p1CreatureAttack;
+    private List<Integer> p2CreatureAttack;
+
+    public DeathDetectorResponseModel(List<Integer> defeatCreatureId, List<Integer> p1CreatureIds, List<Integer> p2CreatureIds, List<Integer> p1CreatureHitPoints, List<Integer> p2CreatureHitPoints, List<Integer> p1CreatureAttack, List<Integer> p2CreatureAttack) {
```
Comment (Kevin22888) [inline src/main/java/use_cases/death_detector_use_case/DeathDetectorResponseModel.java:14]: Just reformatting.
```suggestion
    public DeathDetectorResponseModel(List<Integer> defeatCreatureId, List<Integer> p1CreatureIds,
                                      List<Integer> p2CreatureIds, List<Integer> p1CreatureHitPoints,
                                      List<Integer> p2CreatureHitPoints, List<Integer> p1CreatureAttack,
                                      List<Integer> p2CreatureAttack) {
```

**Review by Kevin22888 — CHANGES_REQUESTED** [review]
Follow-up on @ethanhsu8004's comment and also some reformatting.

File: src/main/java/use_cases/death_detector_use_case/DeathDetectorInteractor.java (hunk 1 above)
Comment (Kaifeng-Li) [reply src/main/java/use_cases/death_detector_use_case/DeathDetectorInteractor.java:32]: nb got it

**Review by Kevin22888 — APPROVED** [review]

**Pull request merged**